src/random.cpp

```diff
@@ -135,6 +135,39 @@ static bool GetHWRand(unsigned char* ent32) {
     return false;
 }
 
+/** Use repeated SHA512 to strengthen the randomness in state. */
+static void Strengthen(CSHA512& hasher, int microseconds)
+{
+    static const unsigned char MARKER_OUTPUT[128] = {'O'};
+    static const unsigned char MARKER_LOCAL[128] = {'L'};
+
+    // Fork the state into a local copy, but write a distinct marker on each side
+    CSHA512 local_hasher = hasher;
+    hasher.Write(MARKER_OUTPUT, sizeof(MARKER_OUTPUT));
+    local_hasher.Write(MARKER_LOCAL, sizeof(MARKER_LOCAL));
+
+    // Hash loop
+    unsigned char buffer[64];
+    int64_t start = GetTimeMicros();
+    do {
+        for (int i = 0; i < 1000; ++i) {
+            local_hasher.Finalize(buffer);
+            local_hasher.Reset();
+            local_hasher.Write(buffer, sizeof(buffer));
+        }
+        // Write benchmark data (into output hasher)
+        int64_t perf = GetPerformanceCounter();
```

Comment:
SeedTimestamp() ?

Reply:
Maybe not because I think you should change the strengthening function to 4-wide sha256d64 in order to increase the "work" done per unit time, and seed timestamp takes a hasher.

Reply:
Done (using `SeedTimestamp()`).

Leaving this for now; switching to SHA256-based code requires some refactoring to make sure there are no initialization order issues.